### ISSUE TITLE
Dynamic data update

### ISSUE THREAD

**Issue by ud-klee** [StephanWagner/svgMap#16]
I figured that the applyData() method can be used to update the map data after the map has been rendered, but the tooltips are not aware of the data change. Can it expose the API to rebuild the tooltips, or use custom events to notify the tooltips on model change? Thanks!

**Comment by ud-klee** [comment]
Aha! Looks like I also need to update `options.data` too... kind of hacky though, it'd be nice if there's a public API for this :)

**Comment by StephanWagner** [comment]
Yeah, the tooltips get the data from the `options` object. Maybe a new method `updateData()` would be a good solution.

**Comment by xaja** [comment]
is there any public api methods? how to update map with new data?

**Comment by StephanWagner** [comment]
At the moment there are none. It's just really what's in the readme or here: https://stephanwagner.me/create-world-map-charts-with-svgmap

It definitely would be a good addition and I hope I'll get around to improve svgmap soon.

**Comment by jjxtra** [comment]
So for now just clear the element children and re-call to create a brand new map?

**Comment by jjxtra** [comment]
I'm using this code and the map does not update, what am I doing wrong?

```
options.data = CreateMapData();
options.data.data.values = newValues;
svgMap.prototype.applyData(options.data);
```

**Comment by arelidev** [comment]
Does anyone find a solution for this? I'm assuming it's not being developed anymore.

**Comment by xaja** [comment]
> Does anyone find a solution for this? I'm assuming it's not being developed anymore.

some thoughts and minimal demo

when you update your map data via `computed` you change the `key` with new unique value, that forces component `<Svgmap>` to rerender with your new data. it is not beautiful, but works

```
<template>
  <div>
    <Svgmap :data="mapData" :key="componentKeyForceRerender"></Svgmap>
  </div>
</template>
<script>
import Svgmap from "@/components/Svgmap.vue";

export default {
  components: {
    Svgmap
  },
  data() {
    return {
      componentKeyForceRerender: 0
    };
  },
  methods: {
    forceRerender() {
      this.componentKeyForceRerender += 1;
    }
  },
  computed: {
    mapData() {
      const result = "your data for map";
      this.forceRerender();
      return result;
    }
  }
};
</script>
```

**Comment by tipsy** [comment]
@ud-klee did you figure out how to do this?

**Comment by tipsy** [comment]
Okay, it's actually quite simple: https://github.com/StephanWagner/svgMap/issues/121#issuecomment-2259281492
